src/pybind/mgr/dashboard/services/cephfs.py

```diff
@@ -133,7 +140,7 @@ def dir_exists(self, path):
         try:
             with self.opendir(path):
                 return True
-        except cephfs.ObjectNotFound:
+        except (cephfs.OSError, cephfs.ObjectNotFound):
```

Comment:
Are we sure we want this behaviour? When there's an error we'll tell the user that the directory doesn't exist... Isn't that a bit misleading?
```suggestion
        except cephfs.ObjectNotFound:
          return False
        except cephfs.OSError as e:
          raise DashboardCephfsError.from_cephfs_error(e) from e
```

Reply:
Oh yes, this is misleading. will fix this. Thanks!

Reply:
The function call stack for this is as follows( in reverse order):  `opendir()` ->`dir_exists()` -> `mk_dirs()`/`rm_dir()` -> `mk_tree`/`rm_tree`.[API]
So here two possibilities can arise while checking if a a directory exists (done by checking if opendir() is successful) while created or removing directories :
 - directory exists but has wrong permission set (+x bit unset) -> thus giving `cephfs.OSError: opendir failed: Permission denied [Errno 13]` in opendir():  dir_exists() should raise an error
  - directory doesn't exist -> giving `cephfs.OSError: opendir failed: No such file or directory [Errno 2]` in opendir(): dir_exists() should return false. As cephfs.ObjectNotFound is derived from cephfs.OSError, it will get caught by the `except cephfs.OSError as err: ` block in opendir() 

(Both  are supposed to  handled  as suggested here: https://github.com/ceph/ceph/pull/44849#discussion_r801499491  in opendir())
 
 This can be tackled in any of the two ways:
 -  adding a check by errno code in dir_exists()
  ```py in src/pybind/mgr/dashboard/services/cephfs.py -> dir_exists()
 except DashboardException as err:
           #for ObjectNotFound
            if(err.code == '2'):
                return False
            else:
                raise err
   ```
   - OR, add  except block for `cephfs.ObjectNotFound` before `except cephfs.OSError` 
   ```py in src/pybind/mgr/dashboard/services/cephfs.py -> opendir()
except cephfs.ObjectNotFound as err:
            raise err
   ```
   What do you think about this?. Both solution look  far from ideal to me.

Reply:
The last one is better IMHO. Exceptions should be handled the closest to where they happen. At that level you'd treat differently the Not Found from the rest of the exceptions, which you can raise as DashboardException. How does that sound?